`Development`: Adjust upload button based on primeng with editing and discarding functionalities

Douglas-Sarah opened the pull request:
<!-- Thanks for contributing to TumApply! Before you submit your pull request, please make sure to check all tasks by putting an x in the [ ] (don't: [x ], [ x], do: [x]). Remove not applicable tasks and do not leave them unchecked -->
<!-- If your pull request is not ready for review yet, create a draft pull request! -->

### Checklist

#### General

<!-- Remove tasks that are not applicable for your PR. Please only put the PR into ready for review, if all relevant tasks are checked! -->
<!-- You only need to choose one of the first two check items: Generally, test on the test servers. -->
<!-- If it's only a small change, testing it locally is acceptable, and you may remove the first checkmark. If you are unsure, please test on the test servers. -->

- [x] I tested **all** changes and their related features with **all** corresponding user types.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://confluence.aet.cit.tum.de/spaces/AP/pages/257786006/Language+Guidelines+Client).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://confluence.aet.cit.tum.de/spaces/AP/pages/256311714/PR+Guidelines).

#### Server

- [x] **Important**: I implemented the changes with a [very good performance](https://confluence.aet.cit.tum.de/spaces/AP/pages/256311718/Performance+Guidelines) and prevented too many (unnecessary) and too complex database calls.
- [x] I **strictly** followed the principle of **data economy** for all database calls.
- [x] I **strictly** followed the [server coding and design guidelines](https://confluence.aet.cit.tum.de/spaces/AP/pages/256311877/Server+Guidelines).
- [x] I documented the Java code using JavaDoc style.

#### Client

- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data (e.g. using paging).
- [x] I **strictly** followed the principle of **data economy** for all client-server REST calls.
- [x] I **strictly** followed the [client coding and design guidelines](https://confluence.aet.cit.tum.de/spaces/AP/pages/256311716/Client+Guidelines).
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://confluence.aet.cit.tum.de/spaces/AP/pages/256311925/Client+Test+Guidelines).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.

### Motivation and Context

Applicants often need to submit multiple supporting documents (e.g., transcripts, references, CVs) as part of their application process. To streamline this experience, we provide an advanced upload component that:

- Supports drag-and-drop and multiple file uploads.
- Allows inline renaming of both new and already-uploaded files.
- Provides immediate feedback on upload progress, size limits, and document states.

This helps reduce friction in the upload process, minimizes support requests, and gives users better control over their submissions.

### Description

Key Capabilities:
- Upload PDFs with drag-and-drop or manual file selection
- Set maximum file size via configurable limits (maxUploadSizeInMb)
- Upload multiple files at once

- Display document state:
    - 🟡 "Pending Upload" (with a yellow icon)
    - 🟢 "Uploaded Successfully" (with a green icon)
- Rename documents inline:
    - New uploads (in-memory `File` objects).
    - Already-uploaded documents (DocumentInformationHolderDTO) using input fields
    - Trigger renaming via blur event that calls serverside renameDocument(...).
- Remove individual documents or clear all uploaded files at once

### Steps for Testing

<!-- Please describe in detail how reviewers can test your changes. Make sure to take all related features and views into account! Below is an example that you can refine. -->

Prerequisites:

1. Log in to TumApply as an applicant
2. Choose "My Applications" and click on "Update"

Test:

3. On Page 2 or 3 of the application form, try to upload a file to the site. (Click "Add" or use drag and drop)
4. Check whether the size is shown correctly
5. Click "Upload" and check whether file is uploaded
6. Change name and check whether name is saved correctly
7. Check whether 'delete' of a single file or multiple files ('Clear'-Button) works
8. Check whether the tooltips work

### Review Progress

<!-- Each PR should be reviewed by at least one other developers. The code, the functionality (= manual test). -->
<!-- The reviewer or author check the following boxes depending on what was reviewed or tested. All boxes should be checked before merge. -->
<!-- You can add additional checkboxes if it makes sense to only review parts of the code or functionality. -->
<!-- When changes are pushed, uncheck the affected boxes. (Not all changes require full re-reviews.) -->

#### Code Review

- [x] Code Review 1

#### Manual Tests

- [x] Test 1

### Screenshots
![image](https://github.com/user-attachments/assets/de50b80c-d5b9-48e4-bdfa-a23d9b5a4a52)

![image](https://github.com/user-attachments/assets/25a66734-dc9f-4dba-9e72-9342e7f16f64)

![image](https://github.com/user-attachments/assets/4f29dc25-e68b-441d-bf9e-4a73b3cf98b2)

MoSchmidt commented on src/main/java/de/tum/cit/aet/core/service/DocumentDictionaryService.java:
```diff
@@ -140,4 +163,19 @@ public ApplicationDocumentIdsDTO getDocumentIdsDTO(Application application) {
+    /**
+     * Updates the name of the document with the given ID.
+     *
+     * @param documentId the ID of the document to rename
```
Can you rename this parameter to documentDictionaryId so it is less confusing? As we have the table documents this suggest we are using the actual documentId

Douglas-Sarah replied:
You are right, fixed 👍

MoSchmidt commented on src/main/webapp/app/playground/document-viewer-playground/document-viewer-playground.component.html:
```diff
@@ -1,5 +1,6 @@
 <div class="w-full mt-4 flex justify-center">
   <div class="h-80 w-80 p-2 self-center rounded-lg border-neutral">
-    <jhi-document-viewer documentDictionaryId="fec0f898-864f-4316-8fcb-72c84d9d0df9" />
+    <!-- TODO -->
```
Why is this commented out and marked as TODO?

MoSchmidt commented on src/main/webapp/app/shared/components/atoms/upload-button/upload-button.component.html:
```diff
@@ -1,44 +1,108 @@
+          }
+          @if (documentIds(); as documentIds) {
+            @for (info of documentIds; track $index) {
+              <tr class="border border-gray-400 leading-0 border-x-0">
```
Please use our color schema instead of the Tailwind colors

Douglas-Sarah replied:
Fixed  👍

MoSchmidt commented on src/main/webapp/app/shared/components/atoms/upload-button/upload-button.component.scss:
```diff
@@ -1,32 +1,41 @@
+.slim-table th,
+.slim-table td {
+  color: var(--p-text-color);
+  padding: 0ren 0.3rem;
```
Typo: pls fix to rem here

MoSchmidt commented on src/main/webapp/content/theming/tumapplypreset.ts:
```diff
@@ -961,4 +961,29 @@ export const TUMApplyPreset = definePreset(Lara, {
+  fileupload: {
+    header: {
+      border: {
+        width: '0px',
```
Nitpick: remove the px here

Douglas-Sarah replied:
Fixed 👍

MoSchmidt commented on src/main/webapp/content/theming/tumapplypreset.ts:
```diff
@@ -961,4 +961,29 @@ export const TUMApplyPreset = definePreset(Lara, {
+      border: {
+        width: '0px',
+      },
+      gap: '0px',
```
Here as well

MoSchmidt reviewed (changes requested):
Left some small comments/questions.

As discussed, I would suggest to remove the "Upload" button. When adding a new document it should directly be send to the server without requireing extra actions from the user

ani-ib commented on src/main/webapp/content/theming/tumapplypreset.ts:
```diff
@@ -961,4 +961,29 @@ export const TUMApplyPreset = definePreset(Lara, {
+  fileupload: {
+    header: {
+      border: {
+        width: '0px',
```
please switch this to rem if possible

Douglas-Sarah replied:
Fixed (no unit is necessary for 0) 👍

ani-ib commented on src/main/webapp/content/theming/tumapplypreset.ts:
```diff
@@ -961,4 +961,29 @@ export const TUMApplyPreset = definePreset(Lara, {
+      border: {
+        width: '0px',
+      },
+      gap: '0px',
```
here as well

Douglas-Sarah replied:
Fixed as well 👍

ani-ib commented on src/main/webapp/i18n/en/global.json:
```diff
@@ -170,9 +170,12 @@
       "uploaded_document": "Uploaded Document for ",
       "uploaded_documents": "Uploaded Documents for ",
-      "instruction": "Click or drag and drop files to here to upload.",
+      "instruction": "Click Add or drag and drop files to here to upload.",
```
Please use single quotation marks, similarly to how you did it for the german translation
```suggestion
      "instruction": "Click on 'Add' or drag and drop files to here to upload.",
```

Douglas-Sarah replied:
The Add button was removed, so the text changed - this is now not necessary anymore

ani-ib reviewed (changes requested):
Thanks for improving the upload button, the changes look very good to me. I added a few small comments.

Furthermore, when I "Add" documents, but don't "Upload" them, then I am unable to press the "x" button to remove them from the list. I am also unable to use the "Alles löschen" button if I haven't uploaded the documents. Clicking these buttons has no functionality, as it doesn't remove the documents from the list.
<img width="762" alt="image" src="https://github.com/user-attachments/assets/7374adc3-0d1a-493a-9be1-394c6bacc8bc" />

Douglas-Sarah commented:
![image](https://github.com/user-attachments/assets/d5459ce0-3e49-49eb-932e-3ad0bb1369e6)
![image](https://github.com/user-attachments/assets/5793fc65-b029-4cbb-8f75-a5e8e010fe1d)

 This is the new design of the button- I will post it on slack as well.

I removed the Upload and the Add Button. Therefore the text will changed as well. The files are uploaded directly, but I adjusted the clear all button as well, so that if the files are not yet uploaded completely, the former "Clear" Button will work on both types of files (uploaded and not yet uploaded).
On click on the button, the files can be uploaded now too.

az108 reviewed (approved):
LGTM
